File: TopEducationApp/score-service/src/main/java/TopEducation/student_scoreservice/services/ScoreService.java
package TopEducation.student_scoreservice.services;

import TopEducation.student_scoreservice.entities.ScoreEntity;
import TopEducation.student_scoreservice.repositories.ScoreRepository;
import lombok.Generated;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service

public class ScoreService {

    @Autowired
    private ScoreRepository scoreRepository;

    // Save a student grade
    public void saveScore(ScoreEntity studentGrade) {
        scoreRepository.save(studentGrade);
    }

    // Delete a student score by rut
    public void deleteScore(String rut) {
        scoreRepository.deleteByRut(rut);
    }

    // Delete all student scores
    @Generated
    public void deleteAllGrades() {
        scoreRepository.deleteAll();
    }

    // Delete all student scores by student RUT
    public void deleteAllScoresByStudentRUT(String scoreRUT) {
        // Get all the scores by student RUT
        List<ScoreEntity> scores = findAllScoresByStudentRUT(scoreRUT);
        // Delete all the scores
        for (ScoreEntity score : scores) {
            deleteScore(score.getScoreRUT());
        }
    }

    // Find by methods

    // Find a student score by id
    public ScoreEntity findScoreById(Long id) {
        return scoreRepository.findById(id).orElse(null);
    }

    // Find all student scores
    public List<ScoreEntity> getAllScores() {
        return (List<ScoreEntity>) scoreRepository.findAll();
    }

    // Find all student scores by student RUT
    public List<ScoreEntity> findAllScoresByStudentRUT(String gradeRUT) {
        return (scoreRepository.findAllScoresByStudentRUT(gradeRUT));
    }
}

File: TopEducationApp/frontend-service/src/services/ScoreService.js
import axios from 'axios';

const SCORE_API_URL = 'http://localhost:8080/scores';

class ScoreService {
    getAllScores() {
        return axios.get(SCORE_API_URL);
    }

    getScoresByRUT(rut) {
        return axios.get(SCORE_API_URL + '/byRUT/' + rut);
    }

    deleteScoreById(scoreId) {
        return axios.delete(SCORE_API_URL + '/delete/' + scoreId);
    }

    deleteAllScores() {
        return axios.delete(SCORE_API_URL + '/deleteAll');
    }

    createScore(score) {
        return axios.post(SCORE_API_URL, score);
    }
}

export default new ScoreService();
